src/notation/internal/notationactioncontroller.cpp

```diff
@@ -36,6 +36,40 @@ static const ActionCode ESCAPE_ACTION_CODE = "escape";
 static const ActionCode UNDO_ACTION_CODE = "undo";
 static const ActionCode REDO_ACTION_CODE = "redo";
 
+template<class P1>
+void NotationActionController::registerAction(const mu::actions::ActionCode& code, void (INotationInteraction::* handler)(P1),
+                                              P1 param1, bool (NotationActionController::* enabler)() const)
+{
+    registerAction(code, [this, handler, param1]()
+    {
+        auto interaction = currentNotationInteraction().get();
+        if (interaction) {
+            (interaction->*handler)(param1);
+        }
+    }, enabler);
+}
```

Comment:
Will move this to the end of the file, I'm pretty sure modern compilers can cope with template definitions being anywhere in the same compilation unit (they didn't 20 years ago!)